packages/babel-plugin-transform-es2015-parameters/src/destructuring.js

```diff
@@ -3,8 +3,12 @@ import * as t from "babel-types";
 export let visitor = {
   Function(path) {
     let params: Array = path.get("params");
-    
-    for (let i = 0; i < params.length; i++) {
+
+    // If there's a rest param, no need to loop through it. Also, we need to
+    // hoist one more level to get `declar` at the right spot.
+    const hoistTweak = t.isRestElement(params[params.length - 1]) ? 1 : 0;
+
+    for (let i = 0; i < params.length - hoistTweak; i++) {
```

Comment:
I also noticed that I used a `const` here and that Babel's codebase doesn't use them. Is there a reason to this?